docs/extension-system/_index.md

```diff
@@ -0,0 +1,146 @@
+---
+title: 'Extension system'
+date: 2024-01-23T00:00:00+00:00
+weight: 60
+geekdocRepo: https://github.com/owncloud/web
+geekdocEditPath: edit/master/docs/extension-system
+geekdocFilePath: _index.md
+geekdocCollapseSection: true
+---
+
+{{< toc >}}
+
+## Concepts and Building Blocks
+
+The ownCloud Web can be extended through various entry points with custom **apps** and **extensions**.
+
+### Distinction between Apps and Extensions
+
+An Application in the context of ownCloud Web is an artifact which can be distributed to and installed by an ownCloud admin.
+It serves two main purposes:
+1. It makes the full app viewport (everything below the top bar) available to the application developer for any custom
+   application code. This includes the ability to define views with routes, navigation items for the left sidebar, and more.
+2. Through the `extensions` key in the application interface the developer can register extensions of any extension type.
+   Those extensions will become available in standardized extension points and for being queried from the extension registry
+   for custom use.
+
+Both parts are optional. This means that an application can be a file editor without any custom extensions, or even contain
+no custom application code at all and only host extensions to be registered in the extension registry, or a combination of both.
+
+### Apps
+
+To get started, define a `src/index.ts`. Below is the most basic example of its content:
+
+```typescript
+// Install '@ownclouders/web-pkg' as a devDependency first (only relevant for types and autocompletion, dependency is already provided by the ownCloud Web at runtime). 
+import {
+  AppWrapperRoute,
+  ApplicationFileExtension,
+  defineWebApplication
+} from '@ownclouders/web-pkg'
+
+
+export default defineWebApplication({
+  setup({ applicationConfig }) {
+    // Here, you have access to the full injection context, meaning you can use all composable that we provide via web-pkg
+
+    // Needs to be unique within all installed extensions in any ownCloud web instance
+    // Should be short, unique and expressive as it gets prefixed on all routes within your application
+    const appId = 'your-extension' 
+
+    // See extensions section below
+    const extensions = [
+        ...
+    ]
+
+    // See details below
+    const navItems = [
+      ...
+    ]
+
+    // See details below
+    const routes = [
+        ...
+    ]
+
+    return {
+      appInfo: {
+        name: $gettext('Your extension'),
+        id: appId,
+        icon: 'aliens', // See https://owncloud.design/#/Design%20Tokens/IconList for available options
+      },
+      extensions,
+      navItems,
+      routes
+    }
+  }
+})
+```
+
+By defining an application via `defineWebApplication` you can provide the following:
+- `appInfo` - the application metadata, which is used to make the application available via the app switcher and the app registry
```

Comment:
Should we link to the `ApplicationInformation` interface here? Or is this supposed to be a quick overview only?

```suggestion
- `appInfo` - the application metadata, which is used to make the application available via the app switcher and the app registry.
```

Reply:
For now only supposed to be a quick overview. We want to come up with more details for applications, when we write about different examples for different apps (e.g. viewers/editors vs. "extension host apps" vs. custom business logic apps, etc).